src/renderers/webgl/WebGLShadowMap.js

```diff
@@ -246,44 +365,78 @@ THREE.WebGLShadowMap = function ( _renderer, _lights, _objects ) {
 
 		}
 
+		_renderer.setViewport( _vector4.x, _vector4.y, _vector4.z, _vector4.w );
+
 		_renderer.resetGLState();
 
 		scope.needsUpdate = false;
 
 	};
 
-	function getDepthMaterial( object, material ) {
+	function getDepthMaterial( object, material, isPointLight, lightPositionWorld ) {
 
 		var geometry = object.geometry;
 
 		var useMorphing = geometry.morphTargets !== undefined && geometry.morphTargets.length > 0 && material.morphTargets;
 		var useSkinning = object instanceof THREE.SkinnedMesh && material.skinning;
 
-		var depthMaterial;
+		var newMaterial;
+
+		var depthMaterial = _depthMaterial;
+		var depthMaterialMorph = _depthMaterialMorph;
+		var depthMaterialSkin = _depthMaterialSkin;
+		var depthMaterialMorphSkin = _depthMaterialMorphSkin;
 
-		if ( object.customDepthMaterial ) {
+		if ( isPointLight ) {
+
+			depthMaterial = _distanceMaterial;
+			depthMaterialMorph = _distanceMaterialMorph;
+			depthMaterialSkin = _distanceMaterialSkin;
+			depthMaterialMorphSkin = _distanceMaterialMorphSkin;
+
+		}
 
-			depthMaterial = object.customDepthMaterial;
+		if ( object.customDepthMaterial || object.customDistanceMaterial ) {
+
+			if ( isPointLight ) {
+
+				newMaterial = object.customDistanceMaterial;
+
+			} else {
+
+				newMaterial = object.customDepthMaterial;
+
+			}
 
 		} else if ( useSkinning ) {
 
-			depthMaterial = useMorphing ? _depthMaterialMorphSkin : _depthMaterialSkin;
+			newMaterial = useMorphing ? depthMaterialMorphSkin : depthMaterialSkin;
 
 		} else if ( useMorphing ) {
 
-			depthMaterial = _depthMaterialMorph;
+			newMaterial = depthMaterialMorph;
 
 		} else {
 
-			depthMaterial = _depthMaterial;
+			newMaterial = depthMaterial;
 
 		}
 
-		depthMaterial.visible = material.visible;
-		depthMaterial.wireframe = material.wireframe;
-		depthMaterial.wireframeLinewidth = material.wireframeLinewidth;
+		newMaterial.visible = material.visible;
+		newMaterial.wireframe = material.wireframe;
+		newMaterial.wireframeLinewidth = material.wireframeLinewidth;
+
+		if ( isPointLight ) {
+
+			if ( newMaterial.uniforms.lightPos ) {
+
+				newMaterial.uniforms.lightPos.value.copy( lightPositionWorld );
+
+			}
+
+		}
```

Comment:
I guess it makes sense to require this uniform on custom depth materials because they will need it?